### PR TITLE
EVMC for fuzzing

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: test/EVMHost.cpp [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+
+evmc::vm& getVM()
+{
+	static unique_ptr<evmc::vm> theVM;
+	if (!theVM)
+	{
+		// TODO make this an option
+		for (auto path: {
+			"deps/lib/libevmone.so",
```
Comment (chriseth) [inline test/EVMHost.cpp:47]: These are the paths were libevmone.so is tried

**Comment by axic** [comment]
I really don't like this unversioned import of evmc. What is the reason against using subtrees?

File: test/EVMHost.cpp [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+			"../deps/lib/libevmone.so",
+			"/usr/lib/libevmone.so",
+			"/usr/local/lib/libevmone.so",
+			// TODO the circleci docker image somehow only has the .a file
+			"/usr/lib/libevmone.a"
+		})
```
Comment (axic) [inline test/EVMHost.cpp:53]: Also please add a `libevmone.dylib` version for macos.

Comment (chriseth) [reply test/EVMHost.cpp:53]: This is just for fuzzing - we'll have to find a much nicer version for general tests anyway.

**Comment by chriseth** [comment]
It's not just unversioned, it is an extremely slimmed down version that removes tons of CMake lines.

**Comment by chriseth** [comment]
Argh, I had this `ar: no archive members specified` before on Linux. Adding `C` as a language helped, but not on MacOS. I guess the actual error here is that we have an object without content, is that right?

**Comment by axic** [comment]
> It's not just unversioned, it is an extremely slimmed down version that removes tons of CMake lines.

Can you explain why is this needed and how evmc could be improved to avoid the need for this? cc @chfast

**Comment by chriseth** [comment]
evmc is almost header-only, I don't see why so much cmake code is needed to make it work.

**Comment by chriseth** [comment]
I tried for some time to configure it as a cmake subdirectory but could not make it work in the end.

**Comment by bshastry** [comment]
This PR is now located at #7069

**Comment by bshastry** [comment]
> Argh, I had this `ar: no archive members specified` before on Linux. Adding `C` as a language helped, but not on MacOS. I guess the actual error here is that we have an object without content, is that right?

The problem afaiu is that we are invoking `ar` against a file (`libevmone.a`) that does not exist. Why `libevmone.a` is not getting built before the invocation of the `ar` command is an independent problem :-)

**Pull request closed without merging**